RealmSwift/Tests/SwiftLinkTests.swift

```diff
@@ -111,4 +111,46 @@ class SwiftLinkTests: TestCase {
 
         XCTAssertEqual(0, owners.count)
     }
+
+    func testLinkingObjectsWithNoPersistedProps() {
+        let realm = realmWithTestPath()
+
+        let target = OnlyComputedProps()
+
+        let source1 = LinkToOnlyComputed()
+        source1.value = 1
+        source1.link = target
+
+        XCTAssertEqual(0, target.backlinks.count, "Linking objects are not available until the object is persisted")
+
+        try! realm.write {
+            realm.add(source1)
+        }
+
+        XCTAssertEqual(1, target.backlinks.count)
+        XCTAssertEqual(source1.value, target.backlinks.first!.value)
+
+        let source2 = LinkToOnlyComputed()
+        source2.value = 2
+        source2.link = target
+
+        XCTAssertEqual(1, target.backlinks.count, "Linking objects to an unpersisted object are not available")
+        try! realm.write {
+            realm.add(source2)
+        }
+
+        XCTAssertEqual(2, target.backlinks.count)
+        XCTAssertTrue(target.backlinks.contains(where: { o in
```

Comment:
Swift automatically provides shorthand argument names to inline closures, so this can be simplified to 
 XCTAssertTrue(target.backlinks.contains(where: { $0.value == 2 }))
 
XCTAssertNotNil(target.backlinks.first { $0.value == 2 })